class.jetpack.php

```diff
@@ -4628,6 +4628,7 @@ function build_connect_url( $raw = false, $redirect = false ) {
 			$url = add_query_arg( $args, Jetpack::api_url( 'authorize' ) );
 		}
 
+		$url = $from ? add_query_arg( 'from', $from, $url ) : $url;
```

Comment:
this is a cute one-liner but I suspect it would be more readable to do something like this:

```php
if ( $from ) {
  $url = add_query_arg( 'from', $from, $url );
}
```